### PR TITLE
[Feature 8] Login Management System

### PR DESCRIPTION
**Pull request by ShuHaoLin2002**
First edition

### CODE REVIEW

**Review by hzhqwq — COMMENTED** [review]
This commit added docstrings and implemented the login UI well, while the homepage UI still needs to improve.

**Review by RitatataVan — COMMENTED** [review]
1. Codes adhere to Clean Architecture and SOLID principles, but you omit to push the LoginRequestModel and LoginResponseModel in your branch. 
2. Testing should be included later to present the evidence of code coverage and relevant documentation.
3. HomePageUI can be improved by setting setBounds, setFont, setHorizontalAlignment, etc. if you want.

File: src/main/java/data_access_storage/UserRepo.java [hunk 1]
```diff
@@ -106,7 +106,8 @@ public boolean existsByEmail(String email) {
      */
     @Override
     public Boolean matchingPassword(String email, String password) {
-        String record_password = accounts.get(email).toString();
```
Comment (cyltaracai) [inline src/main/java/data_access_storage/UserRepo.java:109]: please remove this line of code

File: src/main/java/login_management_system/LoginInteractor.java [hunk 2]
```diff
@@ -32,14 +32,14 @@ public LoginInteractor(UserRepoManager userRepoManager, LoginOutputBoundary logi
     public LoginResponseModel checkValidLogin(LoginRequestModel requestModel){
         // get information from database in order to determine whether the user exists.
         if (!userRepoManager.existsByEmail(requestModel.getName())){
-            return LoginOutputBoundary.prepareFailView("User doesn't exists.");
+            return loginOutputBoundary.prepareFailView("User doesn't exists.");
         }
         // get information from database to determine if the password in record matches user's input.
         else if (!userRepoManager.matchingPassword(requestModel.getName(), requestModel.getPassword())) {
-            return LoginOutputBoundary.prepareFailView("Password doesn't match username.");
+            return loginOutputBoundary.prepareFailView("Password doesn't match username.");
```
Comment (cyltaracai) [inline src/main/java/login_management_system/LoginInteractor.java:39]: The error message could be changed to "You have entered an invalid password."

File: src/main/java/login_management_system/LoginInteractor.java [hunk 3]
```diff
@@ -32,14 +32,14 @@ public LoginInteractor(UserRepoManager userRepoManager, LoginOutputBoundary logi
     public LoginResponseModel checkValidLogin(LoginRequestModel requestModel){
         // get information from database in order to determine whether the user exists.
         if (!userRepoManager.existsByEmail(requestModel.getName())){
-            return LoginOutputBoundary.prepareFailView("User doesn't exists.");
+            return loginOutputBoundary.prepareFailView("User doesn't exists.");
```
Comment (cyltaracai) [inline src/main/java/login_management_system/LoginInteractor.java:35]: The error message could be changed to "You have entered an invalid username."

**Review by cyltaracai — CHANGES_REQUESTED** [review]
Hi Selina, great work. 
No clean architecture violations is found. However, the pull request is missing a few necessary files to make the clean architecture complete such as presenter, and output boundary. I know your have those files locally but I dont see them on this pull request.

**Review by hzhqwq — APPROVED** [review]

**Review by cyltaracai — APPROVED** [review]

**Review by RitatataVan — APPROVED** [review]
Great Work

**Pull request merged**